nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -307,6 +308,7 @@ def model_provider_func(self, pre_process, post_process):
         if self.mcore_gpt:
             model = MCoreGPTModel(
                 config=self.transformer_config,
+                transformer_layer_spec=gpt_layer_with_transformer_engine_spec,
```

Comment:
should we respect the transformer_engine:True/False argument for this? since mcore supports both paths

Reply:
@gshennvm I think transformer_engine param doesn't do anything when mcore_gpt is True